### PR TITLE
AUC support in date

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

File: Date.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 
 
-date::date(std::string init)
+date::date(std::string init, bool AUC)
```
Comment (Zemurin) [inline Date.cpp:7]: no, no, don't invalidate existing code. Rather make a second constructor that takes one bool extra, and the first one internally sets AUC to false by default.

**Review by Zemurin — CHANGES_REQUESTED** [review]

File: Date.cpp (hunk 1 above)
Comment (IhateTrains) [reply Date.cpp:7]: Overloading will work too, but it's not invalidating. This is compatible with current usages, as it's a default param.

Comment (Zemurin) [reply Date.cpp:7]: I see that, and it's true, but when you get code autocompletion in VS, having 2 constructors displays nicely and you can see your options. I'd still prefer 2.

Comment (Idhrendur) [reply Date.cpp:7]: Review comment seconded.

**Review by Idhrendur — CHANGES_REQUESTED** [review]

**Comment by IhateTrains** [comment]
See d6cbf77

**Review by Zemurin — APPROVED** [review]
Now it looks pretty AND enhanced.

**Comment by Zemurin** [comment]
Oh, can you test without the bool, and with other constructors?

**Comment by IhateTrains** [comment]
Don't all the older tests test those?

**Comment by Zemurin** [comment]
> 
> 
> Don't all the older tests test those?

Eh, they do. sorry, didn't see them. :)

**Review by Idhrendur — APPROVED** [review]

**Pull request merged**